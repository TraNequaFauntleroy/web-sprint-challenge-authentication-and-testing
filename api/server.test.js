const server = require('./server')
const request = require('supertest')
const db = require('../data/dbConfig')

beforeAll(async () => {
  await db.migrate.rollback()
  await db.migrate.latest()
})

beforeEach(async () => {
  await db.seed.run()
})

afterAll(async () => {
  await db.destroy()
})

test('sanity', () => {
  expect(true).toBe(true)
})

describe('[POST] /auth/register', () => {
  let res
  beforeEach(async () => {
    res = await request(server).post('/auth/register').send({ name: 'Mariana' })
  })
  it('responds with 201 CREATED', async () => {
    expect(res.status).toBe(201)
  })
  it('causes a user to be added to the db', async () => {
    const user = await db('users')
    expect(user).toHaveLength(3)
  })
  
})


