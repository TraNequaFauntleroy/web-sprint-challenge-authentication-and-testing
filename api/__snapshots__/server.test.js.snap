// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POST] /auth/login responds with the newly created hobbit 1`] = `Object {}`;

exports[`[POST] /auth/register responds with the newly created hobbit 1`] = `Object {}`;

exports[`[POST] /auth/register responds with the newly created user 1`] = `Object {}`;

exports[`[POST] /hobbits responds with the newly created hobbit 1`] = `Object {}`;
